columnphysics/icepack_shortwave.F90

```diff
@@ -2413,6 +2411,8 @@ subroutine compute_dEdd_3bd( &
                gaer = gaer/(waer+puny)
                waer = waer/(taer+puny)
 
+! tcraig, why does the above section exist if taer=waer=gaer=0 below
```

Comment:
Good question!  It's possible that this code is left over from my trying to combine the 3-band and 5-band routines.  If this module is diff'ed with the original icepack_shortwave.F90 in the latest release, does one of the duplicate code sections show up as an addition?

Reply:
The same coding does seem to also be in the Icepack main.  See https://github.com/CICE-Consortium/Icepack/blob/main/columnphysics/icepack_shortwave.F90#L2657 which has exactly the same situation.  So I think it's also in the current code.  One thing to remember is that I don't think we've been testing the modal aerosols at all because they don't work in CICE and the Icepack driver doesn't seem to have all the required options to do those tests either.  So we have no baseline to test against and haven't been exercising that part of the code.

Reply:
It also appears in MPAS-SI in both compute_dEdd (3-band) and compute_dEdd_5bd:  https://github.com/E3SM-Project/E3SM/blob/851fbfbbbdc13915de772519944bc44b4644dfb7/components/mpas-seaice/src/column/ice_shortwave.F90#L4854
@njeffery do you know what's going on here?

Reply:
Apparently this issue entered the code in 2015, in https://github.com/E3SM-Project/CICE-archive/commit/6e8e276fce7eb35338d21d708397cad62ae59d1d# (or maybe another branch feeding this one).  !mfg seems to be Mark Flanner; Hailong Wang is also referenced in the commit message -- this code seems to have come via HiLAT.  
There is a bit of code that was dropped, which used taer etc to calculate g(k), w0(k) and tau(k) for k=0 (lines 2199ff), before taer etc are reset to 0.  k=0 is the surface scattering layer of snow.  [Much of the rest of the compute_dEdd routine calculates g, w0 and tau for the other layers in both snow and ice, which are then used in the solution_dEdd routine.] The missing code is only for tr_aero = true (not tr_zaero). The g, w0 and tau vectors are computed near the end of the tr_aero conditional block, but only for k >= 1, never for k=0.  
This looks like a bug to me.  k=0 values appear to be initialized near the top of the "begin spectral loop", regardless of whether aerosol tracers are turned on, which might mask the impacts of a bug.